Manifest V3 tab-sets (#2272)
MicrosoftDocs/edge-developer: microsoft-edge/extensions-chromium/developer-guide/alternate-distribution-options.md

Extensions that aren't loaded from the Edge Add-ons store are referred to as externally installed extensions. The following are alternate methods of distributing externally installed extensions:

*  Use the Windows registry (Windows only).

> *  Use a preferences JSON file (macOS and Linux). The preferences JSON file would be your **Microsoft Edge extension's CRX ID**.json. The CRX ID is a unique 32 character code which consists of alphabets present at the end of your Microsoft Edge extension's URL. 

Author: ```suggestion
*  Use a preferences JSON file (macOS and Linux). If you use this approach, the name of the preferences JSON file is your Microsoft Edge extension's CRX ID, followed by a `.json` extension. The CRX ID is a unique 32-character code which is the letters that are present at the end of your extension's URL.
```
Engineering, please review this suggested change.

Author: The present comment thread is Outdated.  Latest:
https://github.com/MicrosoftDocs/edge-developer/pull/2272/files

Author: The article has been restructured to avoid premature detail.  Engr approved change.

Extensions that aren't loaded from the Edge Add-ons store are referred to as externally installed extensions. The following are alternate methods of distributing externally installed extensions:

*  Use the Windows registry (Windows only).

> *  Use a preferences JSON file (macOS and Linux). The preferences JSON file would be your **Microsoft Edge extension's CRX ID**.json. The CRX ID is a unique 32 character code which consists of alphabets present at the end of your Microsoft Edge extension's URL. 

Reviewer: "would" maybe should be "is."

Author: PR UI > Files Changed tab isn't showing the present comment.    Its link is 
https://github.com/MicrosoftDocs/edge-developer/pull/2272#discussion_r1007302740

Reviewer 2: Needs engineering review

Author: The wording comes from this commit in the superseded PR 2101:
https://github.com/MicrosoftDocs/edge-developer/pull/2101/commits/f4b3d83e551211468959a2fd3f873b0d8aa931c3

   > [!NOTE]
   > If you want to install an extension from the Chrome Web Store, set the value of `update_url` to `https://clients2.google.com/service/update2/crx`.

> 1. In Microsoft Edge, go to `edge://extensions` and verify that your extension is listed.

Reviewer: replace "and" with "to"

Author: rewrote sentence as two steps

> 1. In Microsoft Edge, go to `edge://extensions` and verify that your extension is listed.

Author: ```suggestion
1. In Microsoft Edge, go to `edge://extensions`, and then verify that your extension is listed.
```

   > [!NOTE]
   > If you want to install an extension from the Chrome Web Store, set the value of `update_url` to `https://clients2.google.com/service/update2/crx`.

1. In Microsoft Edge, go to `edge://extensions`, and then verify that your extension is listed.

<!-- ====================================================================== -->
## Using a preferences JSON file (macOS and Linux)

To distribute your extension by using a preferences JSON file:

1. When using Linux, make sure your `.crx` extension file is available on the machine that the extension will be installed on.  Copy the `.crx` extension file to a local directory, or use a network share that is reachable from the machine.

> 1. Create a JSON file where the name of the file corresponds to the ID of your extension. For example, create a JSON file with the file name `aaaaaaaaaabbbbbbbbbbcccccccccc.json`.<!--30 letters, ok?-->

Author: Engineering, 

The aaa...ccc filename shown is only 30 chars, not 32.  Add two chars to the example string?  Or change "32" to "30" in the following sentence?

Reviewer 3: We should make it 32 characters by adding 2 chars to the example string

Author: In 3 spots in this .md, changed to aaaaaaaabbbbbbbbccccccccdddddddd  (8 chars x 4 = 32 chars)

>    The name of the preferences JSON file is your Microsoft Edge extension's CRX ID, followed by a `.json` extension.  The CRX ID is a unique 32-character code which is the letters that are present at the end of your extension's URL.

Author: Needs Engineering review.  This paragraph was moved from top of file (inherited from the superseded PR 2101) and reworded.

>    The name of the preferences JSON file is your Microsoft Edge extension's CRX ID, followed by a `.json` extension.  The CRX ID is a unique 32-character code which is the letters that are present at the end of your extension's URL.

Author: ```suggestion
   The name of the preferences JSON file must be your Microsoft Edge extension's CRX ID, followed by a `.json` extension.  The CRX ID is a unique 32-character code which is the letters that are present at the end of your extension's URL.
```

Reviewer 3: Looks good to me!

        *  User-specific: `~USERNAME/Library/Application Support/Microsoft Edge/External Extensions/`
        *  For all users: `/Library/Application Support/Microsoft/Edge/External Extensions/`

>         To prevent unauthorized users from installing extensions for all users, make sure your extension preferences file is read-only.

Reviewer 2: Needs eng. review.

Reviewer 3: Looks good

        Also make sure that the following conditions are met:

        *  Every directory in the path is owned by the user root.
        *  Every directory in the path is assigned to the `admin` or `wheel` group.
        *  Every directory in the path isn't `world` writable.
        *  The path must be free of symbolic links.

    *  **Linux**
        *  User-specific: `~/.config/microsoft-edge/External Extensions/`
        *  For all users: `/usr/share/microsoft-edge/extensions/`

> 1. Depending on your scenario, copy the appropriate code that follows, into your preferences JSON file.

Reviewer 2: Needs eng. review

Reviewer 3: Looks good

        }
        ```

    *  Applies to macOS and Linux.  If you install from an `update_url`, specify the update URL in `external_update_url`.

>        Copy the following code into your preferences JSON file when installing from local `.crx` files on Linux only:

Reviewer 2: Needs eng. review

       ```json
       {
           "external_update_url": "http://myhost.com/mytestextension/updates.xml"
       }
       ```

>     *  Copy the following code to your preferences JSON file when installing from the Microsoft Edge Add-ons website on macOS and Linux:

Reviewer 2: Needs eng. review

Reviewer 3: Looks good

Microsoft Edge scans the metadata entries in the registry each time the browser starts, and makes any changes to the externally installed extensions.

To update your extension to a new version, update the `version` string in the extension manifest file, and then update the version in the registry.

You may need to uninstall externally installed extensions, which were installed as part of a bundle of software that was previously installed on the machine.  To uninstall your extension, remove your preferences JSON file or remove the key from the registry.

See also:

> * [Manifest file format for extensions](../getting-started/manifest-format.md)

Reviewer 2: Needs eng. review

Reviewer 3: Looks good
